go/vt/vtgate/planbuilder/select2.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package planbuilder
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// PlanBuilder represents any object that's used to
+// build a plan.
+type PlanBuilder interface{}
+
+// JoinBuilder is used to build a Join primitive.
+type JoinBuilder struct {
+	IsLeft      bool
+	Left, Right PlanBuilder
+}
+
+// RouteBuilder is used to build a Route primitive.
```

Comment:
Put a sentence describing what a Route primitive means (either here, or in the overview comment). It would also be great if you can describe the meaning of each of the members as it's not obvious.
Is "From" supposed to be the pointer to the "AST"? The name is confusing with the "From" clause. Perhaps rename to Ast?